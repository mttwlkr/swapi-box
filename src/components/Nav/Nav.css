.button-nav {
  display: flex;
  height: 80px;
  justify-content: space-evenly;
  align-items: center;
}

.button {
  background: #e6bc42;
  font-family: 'Carrois Gothic', sans-serif;
  font-size: 20px;
  width: 15%;
  height: 40%;
  border: none;
}

