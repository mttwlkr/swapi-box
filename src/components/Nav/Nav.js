import React from 'react'
import Button from '../Button/Button.js'
import './Nav.css'

const Nav = (props) => {
  return (
    <div>
      <Button 
        name='people'
        controlFunction={props.controlFunction}
      />
      <Button 
        name='planets'
        controlFunction={props.controlFunction}
      />
      <Button 
        name='vehicles'
        controlFunction={props.controlFunction}
      />
    </div>
  )
}

export default Nav;