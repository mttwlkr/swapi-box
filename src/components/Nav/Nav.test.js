import React from 'react';
import { shallow, mount } from 'enzyme';
import Nav from './Nav.js';

describe('Nav', () => {

  let mockControlFunction;
  let wrapper;

  beforeEach(() => {
    mockControlFunction = jest.fn()
    wrapper = shallow(<Nav controlFunction={mockControlFunction}/>)    
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke the props control function when a button is clicked', () => {
    wrapper.find('.people-button').simulate('click')
    expect(mockControlFunction).toHaveBeenCalled()
  })
})
