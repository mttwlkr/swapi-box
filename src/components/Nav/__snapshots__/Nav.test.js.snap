// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    controlFunction={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="nav-button people-button"
          name="people"
          onClick={[Function]}
>
          People
</button>,
        <button
          className="nav-button planets-button"
          name="planets"
          onClick={[Function]}
>
          Planets
</button>,
        <button
          className="nav-button vehicles-button"
          name="vehicles"
          onClick={[Function]}
>
          Vehicles
</button>,
      ],
      "className": "button-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "nav-button people-button",
          "name": "people",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "nav-button planets-button",
          "name": "planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "nav-button vehicles-button",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="nav-button people-button"
            name="people"
            onClick={[Function]}
>
            People
</button>,
          <button
            className="nav-button planets-button"
            name="planets"
            onClick={[Function]}
>
            Planets
</button>,
          <button
            className="nav-button vehicles-button"
            name="vehicles"
            onClick={[Function]}
>
            Vehicles
</button>,
        ],
        "className": "button-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "nav-button people-button",
            "name": "people",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "nav-button planets-button",
            "name": "planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "nav-button vehicles-button",
            "name": "vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
